src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -1547,4 +1549,19 @@ protected void loadMoreGuaranteed() throws IOException {
 
     // Can't declare as deprecated, for now, but shouldn't be needed
     protected void _finishString() throws IOException { }
+
+    protected final void createChildArrayContext(final int lineNr, final int colNr) {
+        _streamReadConstraints.validateDepth(++_depth);
```

Comment:
Would be good to measure performance impact; if there is some, we could instead read max depth into local setting, inline check (that is, method local to parser).

Reply:
yes - I'll set up some benchmark testing